crates/client/l1-messages/src/error.rs

```diff
@@ -1,20 +1,37 @@
 use mp_felt::Felt252WrapperError;
 
-#[warn(dead_code)]
-#[derive(Debug, Eq, PartialEq)]
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
+pub enum L1MessagesConfigError {
+    #[error("File with L1 Messages Worker config not found")]
+    FileNotFound(String),
+    #[error("Failed to deserialize L1 Messages Worker Config from config file")]
+    InvalidFile(String),
+}
+
+#[derive(thiserror::Error, Debug, Eq, PartialEq)]
 pub enum L1MessagesWorkerError {
+    #[error("Failed to initialize L1 Messages Worker based on provided Config")]
     ConfigError,
+    #[error("Failed to convert transaction via Runtime API")]
     ConvertTransactionRuntimeApiError,
-    MessageAlreadyProcessed,
+    #[error("Madara Messaging DB Error")]
+    DatabaseError(String),
+    #[error("Message from L1 has been already processed")]
+    L1MessageAlreadyProcessed,
+    #[error("Failed to read/write into Offchain Storage")]
     OffchainStorageError,
+    #[error("Failed to use Runtime API")]
     RuntimeApiError,
+    #[error("Failed to submit transaction into Transaction Pool")]
     SubmitTxError,
+    #[error("Failed to convert L1 Message into Fee")]
     ToFeeError,
-    ToTransactionError,
+    #[error("Failed to convert L1 Message into L2 Transaction")]
+    ToTransactionError(String),
 }
 
 impl From<Felt252WrapperError> for L1MessagesWorkerError {
-    fn from(_e: Felt252WrapperError) -> Self {
-        L1MessagesWorkerError::ToTransactionError
+    fn from(e: Felt252WrapperError) -> Self {
+        L1MessagesWorkerError::ToTransactionError(format!("{:?}", e))
```

Comment:
You can just store `e` directly rather than formatting it. And relly on your Error impl on `L1MessagesWorkerError` to do the formating

Reply:
Store the error in the error strut, and let the caller decide what he wants to do with it. Maybe he wants to match it and not print it. By converting to string here you make his job more difficult